ral-facilities/datagateway-api
# Define Endpoints for Search API

This PR will close #257

## Description
This PR defines endpoints which need to be implemented for the search API.

I've used the `endpoint=` argument for `add_resource()` to prevent naming collisions between endpoints. This was happening with endpoints like `/datasets` (DataGateway API endpoint) and `/search_api/datasets` even though they're actually different endpoints. Using `endpoint=` fixed this.

#256 isn't implemented yet so the [search API endpoints are always created](https://github.com/ral-facilities/datagateway-api/blob/bc1d09d69aeb2378a7ec65a0f0350de3bb96deb5/datagateway_api/src/api_start_utils.py#L206). Once the new config has been implemented, endpoint creation should respect the config and only create them when they are wanted by the user. I have also [hardcoded the extension](https://github.com/ral-facilities/datagateway-api/blob/bc1d09d69aeb2378a7ec65a0f0350de3bb96deb5/datagateway_api/src/api_start_utils.py#L208) which should be changed to look up the relevant config value once #256 is closed.

To remember to resolve both TODOs, I have opened #283 to tackle these issues once #256 and #257 have been completed.

## Testing Instructions
Run the API on this branch and ensure that you can send requests to each of the newly defined endpoints (e.g. `/search_api/datasets`) which return a 200 and a response body of `null`. 

I was a little undecided on whether this needs a version bump or not. I've gone for feature because these changes do add in new endpoints, even if they are unimplemented. I would welcome an alternative view on this though, I'm not sure if a different (or even no) version bump is more appropriate. 


## Agile Board Tracking
Connect to #257
